public/components/datasources/components/new/configure_datasource.tsx

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiPage,
+  EuiPageBody,
+  EuiSpacer,
+  EuiTitle,
+  EuiText,
+  EuiLink,
+  EuiButton,
+  EuiSteps,
+  EuiPageSideBar,
+  EuiPanel,
+  EuiFormRow,
+  EuiFieldText,
+  EuiBottomBar,
+  EuiButtonEmpty,
+  EuiTextArea,
+} from '@elastic/eui';
+import React, { useCallback, useEffect, useState } from 'react';
+import { ConfigureS3Datasource } from './configure_s3_datasource';
+import { coreRefs } from '../../../../../public/framework/core_refs';
+import { DATACONNECTIONS_BASE } from '../../../../../common/constants/shared';
+import { ReviewS3Datasource } from './review_s3_datasource_configuration';
+import { useToast } from '../../../../../public/components/common/toast';
+
+interface ConfigureDatasourceProps {
+  type: string;
+}
+
+export function Configure(props: ConfigureDatasourceProps) {
+  const { type } = props;
+  const { http } = coreRefs;
+  const { setToast } = useToast();
+
+  const [name, setName] = useState('');
+  const [details, setDetails] = useState('');
+  const [arn, setArn] = useState('');
+  const [store, setStore] = useState('');
+  const [roles, setRoles] = useState<Array<{ label: string }>>([]);
+  const [selectedQueryPermissionRoles, setSelectedQueryPermissionRoles] = useState<
+    Array<{ label: string }>
+  >([]);
+  const [page, setPage] = useState<'configure' | 'review'>('configure');
+  const ConfigureDatasourceSteps = [
+    {
+      title: 'Step 1',
+      children: (
+        <EuiText>
+          <p>Configure Connection</p>
+        </EuiText>
+      ),
+    },
+    {
+      title: 'Step 2',
+      children: (
+        <EuiText>
+          <p>Review Configuration</p>
+        </EuiText>
+      ),
+    },
+  ];
+
+  useEffect(() => {
+    http!.get('/api/v1/configuration/roles').then((data) =>
+      setRoles(
+        Object.keys(data.data).map((key) => {
+          return { label: key };
+        })
+      )
+    );
+  }, []);
+
+  const ConfigureDatasource = (configurationProps: { datasourceType: string }) => {
+    const { datasourceType } = configurationProps;
+    switch (datasourceType) {
+      case 'S3':
+        return (
+          <ConfigureS3Datasource
+            currentName={name}
+            currentDetails={details}
+            setNameForRequest={setName}
+            setDetailsForRequest={setDetails}
+            currentArn={arn}
+            setArnForRequest={setArn}
+            currentStore={store}
+            setStoreForRequest={setStore}
+            roles={roles}
+            selectedQueryPermissionRoles={selectedQueryPermissionRoles}
+            setSelectedQueryPermissionRoles={setSelectedQueryPermissionRoles}
+          />
+        );
+      default:
+        return <></>;
+    }
+  };
+
+  const ReviewDatasourceConfiguration = (configurationProps: { datasourceType: string }) => {
```

Comment:
can type be arbitrary string? if not might be helpful to narrow down to `datasourceType: 'S3' | 'xxx'`

Reply:
Done in [680763b](https://github.com/opensearch-project/dashboards-observability/pull/1049/commits/680763b566a24a70d93f79214331182f495ed291)